src/components/common/CourseReviewBox.tsx

```diff
@@ -241,9 +252,11 @@ const CourseReviewBoxContent = ({
   const notifyUpdate = () => toast(REVIEW_SUCCESS.updated);
 
   const profId =
-    selectedProf === -1 || selectedProf === profsTeaching.length
+    selectedProf === -1 || selectedProf === 0
       ? null
-      : profsTeaching[selectedProf].prof.id;
+      : selectedProf <= profsTeaching.length
+      ? profsTeaching[selectedProf - 1].prof.id
+      : allProfs[selectedProf - profsTeaching.length - 1].id;
```

Comment:
nit: this math could be confusing later, can we first create a variable to hold all profs by concating `profsTeaching` and `filterDuplicateProfs`, and then just render that directly?

We could even just make it a list of `{ label: string, id: string | null }`, where the first element is the null ID + "My Professor Isn't Here" label, and then the rest are just the prof names mapped to the IDs. Then in the options, we just render the names but the indexes will be consistent with the ID